core/src/main/java/ai/timefold/solver/core/impl/domain/variable/cascade/CascadingUpdateShadowVariableDescriptor.java

```diff
@@ -137,8 +137,39 @@ public void linkVariableDescriptors(DescriptorPolicy descriptorPolicy) {
                 MemberAccessorFactory.MemberAccessorType.REGULAR_METHOD, null, descriptorPolicy.getDomainAccessType());
     }
 
-    public void linkVariableDescriptorToSource(CascadingUpdateShadowVariable listener) {
-        var sourceDescriptor = entityDescriptor.getShadowVariableDescriptor(listener.sourceVariableName());
+    public void linkVariableDescriptorToSource(CascadingUpdateShadowVariable shadowVariable) {
+        var noEmptySources = Arrays.stream(shadowVariable.sourceVariableNames())
```

Comment:
Nitpick: Should be called `nonEmptySources`; `noEmptySources` would be for a boolean, not a list.